libs/datamodel/core/src/transform/ast_to_dml/db/attributes/relation.rs

```diff
@@ -63,20 +63,120 @@ pub(super) fn validate_ignored_related_model(field: RelationFieldWalker<'_, '_>,
     let related_model = field.related_model();
     let model = field.model();
 
-    if related_model.attributes().is_ignored && !field.attributes().is_ignored && !model.attributes().is_ignored {
-        let ast_model = model.ast_model();
-        let ast_related_model = related_model.ast_model();
-        let ast_field = field.ast_field();
-
-        let message = format!(
-            "The relation field `{}` on Model `{}` must specify the `@ignore` attribute, because the model {} it is pointing to is marked ignored.",
-            ast_field.name(), ast_model.name(), ast_related_model.name()
-        );
-
-        errors.push(DatamodelError::new_attribute_validation_error(
-            &message,
-            "ignore",
-            ast_field.span,
-        ));
+    if !related_model.attributes().is_ignored || field.attributes().is_ignored || model.attributes().is_ignored {
+        return;
+    }
+
+    let ast_model = model.ast_model();
+    let ast_related_model = related_model.ast_model();
+    let ast_field = field.ast_field();
+
+    let message = format!(
+        "The relation field `{}` on Model `{}` must specify the `@ignore` attribute, because the model {} it is pointing to is marked ignored.",
+        ast_field.name(), ast_model.name(), ast_related_model.name()
+    );
+
+    errors.push(DatamodelError::new_attribute_validation_error(
+        &message,
+        "ignore",
+        ast_field.span,
+    ));
+}
+
+/// Some connectors expect us to refer only unique fields from the foreign key.
+pub(super) fn validate_references_unique_fields(
+    field: RelationFieldWalker<'_, '_>,
+    connector: &dyn Connector,
+    errors: &mut Vec<DatamodelError>,
+) {
+    if field.referenced_fields().len() == 0 || !errors.is_empty() {
+        return;
+    }
+
+    if connector.supports_relations_over_non_unique_criteria() {
+        return;
+    }
+
+    let references_unique_criteria = field.related_model().unique_criterias().any(|criteria| {
+        let mut criteria_field_names: Vec<_> = criteria.fields().map(|f| f.name()).collect();
+        criteria_field_names.sort_unstable();
+
+        let mut references_sorted: Vec<_> = field.referenced_fields().map(|f| f.name()).collect();
+        references_sorted.sort_unstable();
+
+        criteria_field_names == references_sorted
+    });
+
+    if references_unique_criteria {
+        return;
+    }
+
+    errors.push(DatamodelError::new_validation_error(
+        &format!(
+            "The argument `references` must refer to a unique criteria in the related model `{}`. But it is referencing the following fields that are not a unique criteria: {}",
+            field.related_model().ast_model().name(),
+            field.referenced_fields().map(|f| f.ast_field().name()).join(", ")
+        ),
+        field.ast_field().span
+    ));
+}
```

Comment:
Should that happen at this stage, or after we inferred the `references`, in case it is implicit?

Reply:
For now it doesn't break any tests, and there are quite a many testing this functionality. But I agree, we should validate these later in the pipeline.